common/sketch/src/main/java/org/apache/spark/util/sketch/CountMinSketchImpl.java

```diff
@@ -325,27 +321,43 @@ public void writeTo(OutputStream out) throws IOException {
   }
 
   public static CountMinSketchImpl readFrom(InputStream in) throws IOException {
+    CountMinSketchImpl sketch = new CountMinSketchImpl();
+    sketch.readFrom0(in);
+    return sketch;
```

Comment:
Hm, I prefer not, because then we'll have to make the no-arg constructor package private instead of private.